contracts/global-extensions/GlobalOptimisticAuctionRebalanceExtension.sol

```diff
@@ -0,0 +1,328 @@
+            address _quoteAsset;    // Address of the quote asset used in the rebalance.
+            address[] _oldComponents; // Addresses of existing components in the SetToken.
+            address[] _newComponents; // Addresses of new components to be added.
+            AuctionExecutionParams[]  _newComponentsAuctionParams; // AuctionExecutionParams for new components, indexed corresponding to _newComponents.
+            AuctionExecutionParams[]  _oldComponentsAuctionParams; // AuctionExecutionParams for existing components, indexed corresponding to the current component positions. Set to 0 for components being removed.
+            bool _shouldLockSetToken; // Indicates if the rebalance should lock the SetToken.
+            uint256 _rebalanceDuration; // Duration of the rebalance in seconds.
+            uint256 _positionMultiplier; // Position multiplier at the time target units were calculated.
+        }
+
+        struct ProductSettings{
+            OptimisticRebalanceParams optimisticParams;
+            bytes32 rulesHash;
+        }
+
+    /* ============ State Variables ============ */
+    
+    // IAuctionRebalanceModuleV1 public immutable auctionModule;  // AuctionRebalanceModuleV1
+
+    mapping (ISetToken=>ProductSettings) public productSettings; // Mapping of quote asset to ProductSettings
+    mapping(bytes32 => bytes32) public assertionIds; // Maps proposal hashes to assertionIds.
+    mapping(bytes32 => bytes32) public proposalHashes; // Maps assertionIds to proposal hashes.
+    mapping (bytes32=>ISetToken) public assertedProducts; // Maps assertionIds to ISetToken
+
+    // Keys for assertion claim data.
+    bytes public constant PROPOSAL_HASH_KEY = "proposalHash";
+    bytes public constant RULES_KEY = "rulesIPFSHash";
+
+
+    /* ============ Constructor ============ */
+
+    constructor(AuctionExtensionParams memory _auctionParams, OptimisticRebalanceParams memory _optimisticParams) public GlobalAuctionRebalanceExtension(_auctionParams.managerCore, _auctionParams.auctionModule) {
+    
+    }
+
+    /* ============ External Functions ============ */
+
+    /**
+    * @dev OPERATOR ONLY: sets product settings for a given quote asset
+    * @param _product Address of the SetToken to set rules and settings for.
+    * @param _optimisticParams OptimisticRebalanceParams struct containing optimistic rebalance parameters.
+    * @param _rulesHash bytes32 containing the ipfs hash rules for the product.
+    */
+    function setProductSettings(
+        ISetToken _product,
+        OptimisticRebalanceParams memory _optimisticParams,
+        bytes32 _rulesHash
+    )
+        external
+        onlyOperator(_product)
+    {
+        productSettings[_product] = ProductSettings({
+            optimisticParams: _optimisticParams,
+            rulesHash: _rulesHash
+        });
+    }
+
+     /**
+    * @dev OPERATOR ONLY: sets product settings for a given quote asset
```

Comment:
```suggestion
    * @dev OPERATOR ONLY: sets product settings for a given set token
```

Reply:
This description seems to be generally wrong as this function proposes the rebalance claim to the oracle. 
Also right now this function **CAN BE CALLED BY ANYONE** and **NOT LIMITED TO OPERATOR ONLY**.
(This might be inteded in which case we should remove the misleading claim from the docstring)